formfyxer/lit_explorer.py

```diff
@@ -347,90 +352,97 @@ def cluster_screens(fields=[],damping=0.7):
 
     return screens
 
-# Get the text content of a pdf.
-
-def read_pdf (file):
-    try:
-        pdfFile = PyPDF2.PdfFileReader(open(file, "rb"))
-        if pdfFile.isEncrypted:
-            try:
-                pdfFile.decrypt('')
-                #print ('File Decrypted (PyPDF2)')
-            except:
-                #
-                # This didn't go so well on my Windows box so I just ran this in the pdf folder's cmd:
-                # for %f in (*.*) do copy %f temp.pdf /Y && "C:\Program Files (x86)\qpdf-8.0.2\bin\qpdf.exe" --password="" --decrypt temp.pdf %f
-                #
-                
-                command="cp "+file+" tmp/temp.pdf; qpdf --password='' --decrypt tmp/temp.pdf "+file
-                os.system(command)
-                #print ('File Decrypted (qpdf)')
-                #re-open the decrypted file
-                pdfFile = PyPDF2.PdfFileReader(open(file, "rb"))
-        text = ""
-        for page in pdfFile.pages:
-            text = text + " " + page.extractText()
-        text = reCase(text)
-        text = re.sub("(\.|,|;|:|!|\?|\n|\]|\))","\\1 ",text)
-        text = re.sub("(\(|\[)"," \\1",text)
-        text = re.sub(" +"," ",text)
-        return text
-    except:
-        return ""
-
-# Read in a pdf, pull out basic stats, attempt to normalize its form fields, and re-write the file with the new fields (if `rewrite=1`). 
+def get_existing_pdf_fields(in_file: Union[str, Path, BinaryIO, pikepdf.Pdf]) -> Iterable:
+    """
+    Use PikePDF to get fields from the PDF
+    """
+    if isinstance(in_file, pikepdf.Pdf):
+      in_pdf = in_file
+    else:
+      in_pdf = pikepdf.Pdf.open(in_file)
+    return [{'type': field.FT, 'var_name': str(field.T), 'all': field} for field in in_pdf.Root.AcroForm.Fields]
+
+
+def unlock_pdf_in_place(in_file:str):
+    """
+    Try using pikePDF to unlock the PDF it it is locked. This won't work if it has a non-zero length password.
+    """
+    if not isinstance(in_file, str):
+        return
+    pdf_file = pikepdf.open(in_file, allow_overwriting_input=True)
+    if pdf_file.is_encrypted:
+        pdf_file.save(in_file)
+
+def cleanup_text(text:str)->str:
+    """
+    Apply cleanup routines to text to provide more accurate readability statistics.
+    """
+    # Replace \n with .
+    text = re.sub(r"(\n|\r)+", ". ", text)
+    # Replace non-punctuation characters with " "
+    text = re.sub(r"[^\w.,;!?@'\"“”‘’'″‶ ]", " ", text)
+    # _ is considered a word character, remove it
+    text = re.sub(r"_+", " ", text)
+    # Turn : into . (so fields are treated as one sentence)
+    text = re.sub(r":", ".", text)
```

Comment:
I'm not sure this is right, though it's probably fine for now. The closest thing I saw that deals with this sort  of thing in the literature is that for multiple choice answers, all of the choices are generally read as the same sentence (I'll have to find the citation for that though, I may not have access to it right now). 

I would guess the equivalent would be to read all of the field labels in a group as one sentence (i.e., First Name, Last Name, vs First Name, Middle Name, Last Name, and Suffix), but that's a bit more complicated.

Reply:
I can wait to merge until we figure out the right choice there. I'm not sure that field labels are similar to multiple choice options, but it would be interesting to read the reasons for people making that choice. Maybe because the readability stats are too inaccurate when applied to small pieces of text--but that's basically our thesis justifying a new measure.